### PR TITLE
Add a LINT option to CMake

### PR DESCRIPTION
**Pull request by tiennou**
This makes the style checks from CI available locally, and also enables an "automatic" linting mode (that makes Xcode report styling issues).

### CODE REVIEW

**Comment by tiennou** [comment]
Bump ? As it's mainly developer-oriented, and helps to enforce the style guide, could this be merged ? Use `cmake -DLINT=ON` to enable.

**Comment by willco007** [comment]
Any chance you could add documentation for this flag in the `INSTALL_CMAKE` file? Thanks!

**Comment by LooTKinGz** [comment]
Creating new repository for Blink

**Comment by tiennou** [comment]
@willco007 Sure, I've added a quick note to the documentation, and rebased against master.

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -170,6 +170,10 @@ add_target_to_copy_dependencies(
   BEFORE_TARGETS ${TEST_TARGETS})
 
 
+if(LINT)
+  add_dependencies(libssh2_clar lint)
```
Comment (willco007) [inline tests/CMakeLists.txt:174]: Is this needed or is it part of the clar test case PR?

**Comment by willco007** [comment]
@tiennou My comment on April 30 was never commented on, did you have a comment on that line of code?

**Comment by tiennou** [comment]
Hi @willco007 ! I've rebased and removed the references to clar from the CMakeLists files. Sadly Travis seems to be stuck, so the build can't go 👌.

**Pull request merged**